tools/test_containers_compose

```diff
@@ -2,11 +2,54 @@
 
 set -euo pipefail
 
-for i in webui worker; do
+wait_until(){
+  command=$1
+  timeout=$2
+
+  i=1
+  until eval "$command"
+  do
+      ((i++))
+      if [[ $i -gt $timeout ]]; then
+          echo "Command '$command' aborted because of timeout"
+          exit 1
+      fi
+      sleep 1
+  done
+}
+
+setup_containers(){
   (
-    cd container/$i/ &&
-     for retry in {2..0}; do sudo docker-compose build && break; echo "Remaining retries $retry"; done &&
-    sudo docker-compose up -d &&
+    for retry in {2..0}; do sudo docker-compose build && break; echo "Remaining retries $retry"; done &&
+    sudo MOJO_CLIENT_DEBUG=1 docker-compose up -d &&
     (docker-compose ps --services --filter status=stopped | grep "^[[:space:]]*$") || (docker-compose logs; sudo docker-compose ps; exit 1)
   ) || exit 1;
-done
+}
+
+test_webui(){
+  (
```

Comment:
I'm wondering why one would spawn a sub-shell all the time. This doesn't make the code more readable.

Reply:
I have to use this to prevent to contextualize the workspace for each test and preventing the use of pushd and popd t (or using cd..)
I remember en in previous review that we prefer sub-shells